src/watchdog.js

```diff
@@ -64,6 +64,14 @@ export default class Watchdog {
 		 */
 		this._crashNumberLimit = typeof config.crashNumberLimit === 'number' ? config.crashNumberLimit : 3;
 
+		/**
+		 * Returns the result of `Date.now()` call. It can be overridden in tests to mock time as the popular
+		 * approaches like `sinon.useFakeTimers()` does not work well with error handling.
+		 *
+		 * @protected
+		 */
+		this._now = () => Date.now();
```

Comment:
Why not:
```suggestion
		this._now = Date.now;
```
?

Reply:
I'm afraid of such functions because sometimes these methods don't work if they use `this` internally. But it works, so I'll change it.